pyglmnet/pyglmnet.py

```diff
@@ -773,6 +788,16 @@ def fit(self, X, y):
             # Compute and save loss if callbacks are requested
             if callable(self.callback):
                 self.callback(beta)
+        else:
+            # Warn if it hit max_iter without converging
+            self.converged = False
+            self.n_iter += t + 1
+            msg = ('\t'
+                   'Failed to converge after {0:d} iterations.'
+                   ' Last convergence metric was {1:f}'
+                   ' and convergence threshold {2:f}.'
+                   ).format(self.n_iter, convergence_metric, tol)
```

Comment:
can you add a test which checks that the warning is indeed raised?

Reply:
Do you have a preferred tool for testing log outputs? I don't see any other log tests in the package that I could base that off of, so I'd probably build something using testfixtures unless you have something else you like.

Reply:
I would have used `pytest.warns`: https://docs.pytest.org/en/latest/warnings.html#warns